src/framework/control.ml

```diff
@@ -275,7 +275,11 @@ struct
               (try funs := Cilfacade.find_varinfo_fundec f :: !funs with Not_found -> ())
             | _ -> ()
           );
-          Spec.assign {ctx with local = st} lval exp
+          let res = Spec.assign {ctx with local = st} lval exp in
+          (* Needed for privatizations (e.g. None) that do not side immediately *)
+          let res' = Spec.sync {ctx with local = res} `Normal in
+          if M.tracing then M.trace "global_inits" "\t\t -> state:%a\n" Spec.D.pretty res;
+          res'
```

Comment:
Wouldn't it be enough to only `sync` once after all the `assign`s, instead of after each `assign`. With a large number of global initializers, this could be wasteful. Also each `sync` in those privatizations iterates over the entire state again. Doing that once should be more optimal.

Reply:
No, that would not suffice. With none privatization, the analysis looks up the value via `getg`, so `sync` has to happen right away, for the correct result to be returned (in case two initializers refer to to the same global, e.g., a struct)

Reply:
Also, the state should be very small, so iterating over it will not be too expensive I think. One could think about only doing this for `earlyglobs`, but even if we do this, it sounds a bit cumbersome.